keras/backend/theano_backend.py

```diff
@@ -1997,7 +1997,75 @@ def separable_conv1d(x, depthwise_kernel, pointwise_kernel, strides=1,
 
 def separable_conv2d(x, depthwise_kernel, pointwise_kernel, strides=(1, 1),
                      padding='valid', data_format=None, dilation_rate=(1, 1)):
-    raise NotImplementedError
+    """2D convolution with separable filters.
+
+    # Arguments
+        x: input tensor
+        depthwise_kernel: convolution kernel for the depthwise convolution.
+        pointwise_kernel: kernel for the 1x1 convolution.
+        strides: strides tuple (length 2).
+        padding: string, `"same"` or `"valid"`.
+        data_format: string, `"channels_last"` or `"channels_first"`.
+        dilation_rate: tuple of integers,
+            dilation rates for the separable convolution.
+
+    # Returns
+        Output tensor.
+
+    # Raises
+        ValueError: if `data_format` is neither `"channels_last"` or `"channels_first"`.
+    """
+    if data_format is None:
+        data_format = image_data_format()
+    if data_format not in {'channels_first', 'channels_last'}:
+        raise ValueError('Unknown data_format ', data_format)
+
+    if hasattr(x, '_keras_shape'):
```

Comment:
I'm thinking we should probably modify `int_shape` to return `None` if the static shape cannot be computed. This would avoid doing the `hasattr` checks. What do you think?

Reply:
@fchollet, There are no restrictions in "the layers" for Theano. Just implementing the backend `separable_conv2d` enable us to perform the following codes:

```python
%env KERAS_BACKEND=theano
import numpy as np
from keras.layers import SeparableConv2D
from keras.models import Sequential
model = Sequential()
model.add(SeparableConv2D(10, (3, 3), input_shape=(5, 5, 3)))
model.predict(np.random.random((1, 5, 5, 3)))
```

And I think the modification of `int_shape` is good idea!